src/worker/activities.rs

```diff
@@ -119,6 +122,13 @@ impl WorkerActivityTasks {
         self.poller.notify_shutdown();
     }
 
+    /// Wait for all outstanding activity tasks to finish
+    pub(crate) async fn wait_all_finished(&self) {
+        while !self.outstanding_activity_tasks.is_empty() {
+            self.complete_notify.notified().await
+        }
```

Comment:
How about this along with the next suggestion?
```suggestion
        self.complete_notify.notified().await
```

Reply:
Or maybe even with a oneshot channel

Reply:
In principle you could use this function anywhere to wait for all current oustanding tasks to complete (not just while shutting down) which is nice. Oneshot would prevent that.